mslib/msui/mscolab.py

```diff
@@ -653,12 +656,15 @@ def get_recent_pid(self):
             "token": self.token
         }
         r = requests.get(self.mscolab_server_url + '/projects', data=data)
-        _json = json.loads(r.text)
-        projects = _json["projects"]
-        p_id = None
-        if projects:
-            p_id = projects[-1]["p_id"]
-        return p_id
+        if r.text != "False":
+            _json = json.loads(r.text)
+            projects = _json["projects"]
+            p_id = None
+            if projects:
+                p_id = projects[-1]["p_id"]
+            return p_id
+        else:
+            show_popup(self, "Error", "Session expired, new login required")
```

Comment:
Does this do the disconnect also?

Reply:
we can discuss how we can logout, the connection is still active and data can be saved when a new login is done.